mailbox/plugin/deleted-messages-vault/src/test/java/org/apache/james/vault/DeletedMessageVaultHookTest.java

```diff
@@ -0,0 +1,224 @@
+    }
+
+    @Test
+    void notifyDeleteShouldAppendMessageToOwnerVaultWhenSharedUserDeleteMessageInSharingMailbox() throws Exception {
+        Mailbox mailboxOne = testingData.createMailbox(MAILBOX_ALICE_ONE, aliceSession);
+        MessageManager aliceMessageManager = testingData.createMessageManager(mailboxOne, aliceSession);
+        MessageManager bobMessageManager = testingData.createMessageManager(mailboxOne, bobSession);
+        ComposedMessageId composedMessageId = appendMessage(aliceMessageManager);
+        MessageId messageId = composedMessageId.getMessageId();
+
+        mailboxManager.setRights(MAILBOX_ALICE_ONE,
+            MailboxACL.EMPTY.apply(MailboxACL.command()
+                .forUser(BOB_ADDRESS)
+                .rights(MailboxACL.Right.Lookup, MailboxACL.Right.Read, MailboxACL.Right.DeleteMessages, MailboxACL.Right.PerformExpunge)
+                .asAddition())
+            ,aliceSession);
+
+        DeletedMessage deletedMessage = buildDeletedMessage(ImmutableList.of(mailboxOne.getMailboxId()), messageId, ALICE);
+        bobMessageManager.delete(ImmutableList.copyOf(bobMessageManager.search(searchQuery, bobSession)), bobSession);
+
+        assertThat(messageVault.search(ALICE, Query.ALL).collectList().block().get(0))
+            .isEqualTo(deletedMessage);
+    }
+
+    @Test
+    void notifyDeleteShouldAppendMessageToSharedUserVaultWhenSharedUserDeleteMessageAfterMoveToAnotherMailbox() throws Exception {
+        Mailbox mailboxOne = testingData.createMailbox(MAILBOX_ALICE_ONE, aliceSession);
+        Mailbox mailboxTwo = testingData.createMailbox(MAILBOX_BOB_ONE, bobSession);
+        MessageManager aliceMessageManager = testingData.createMessageManager(mailboxOne, aliceSession);
+        MessageManager bobMessageManager = testingData.createMessageManager(mailboxTwo, bobSession);
+        ComposedMessageId composedMessageId = appendMessage(aliceMessageManager);
+        MessageId messageId = composedMessageId.getMessageId();
+
+        mailboxManager.setRights(MAILBOX_ALICE_ONE,
+            MailboxACL.EMPTY.apply(MailboxACL.command()
+                .forUser(BOB_ADDRESS)
+                .rights(MailboxACL.Right.Lookup, MailboxACL.Right.Read, MailboxACL.Right.DeleteMessages, MailboxACL.Right.PerformExpunge)
+                .asAddition())
+            ,aliceSession);
+
+        messageIdManager.setInMailboxes(messageId, ImmutableList.of(mailboxTwo.getMailboxId()), bobSession);
+
+        DeletedMessage deletedMessage = buildDeletedMessage(ImmutableList.of(mailboxTwo.getMailboxId()), messageId, BOB);
+        bobMessageManager.delete(ImmutableList.copyOf(bobMessageManager.search(searchQuery, bobSession)), bobSession);
+
+        assertThat(messageVault.search(BOB, Query.ALL).collectList().block().get(0))
+            .isEqualTo(deletedMessage);
+    }
+
+    @Test
+    void notifyDeleteShouldAppendMessageToOwnerVaultWhenSharedUserDeleteMessageAfterCopyToAnotherMailbox() throws Exception {
```

Comment:
> Bob shared mbx1 containing m1 to Alice
Alice copies m1 in mbx2 belonging to her
Alice deletes m1 in mbx2
m1 should be in Bob vault

IMO, I think m1 should be in Alice. Please check my test here.

Reply:
This test is cool.

Can we get the exact same, but checking Alice vault?

 ```
Bob shared mbx1 containing m1 to Alice
Alice copies m1 in mbx2 belonging to her
Alice deletes m1 in mbx2
m1 should be in Alice vault
```

Reply:
Also I believe we need a check :

```
This test is cool.

Can we get the exact same, but checking Alice vault?

 ```
Alice shared mbx1 containing m1 to Bob
Bob deletes m1 in mbx1
m1 should not be in Bob vault
```

Reply:
added